core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -357,7 +357,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   def setLogLevel(logLevel: String) {
     val validLevels = Seq("ALL", "DEBUG", "ERROR", "FATAL", "INFO", "OFF", "TRACE", "WARN")
-    if (!validLevels.contains(logLevel)) {
+    // let's allow lowcase or mixed case too
+    if (!validLevels.contains(logLevel.toUpperCase(Locale.ENGLISH))) {
```

Comment:
@srowen Thanks! Oh. I see what you mean. I can move the validLevels as a set as a static member. And I can also store the `logLevel.toUpperCase` to a val and pass it to log4j. The reason I was passing `logLevel` as is is because `org.apache.log4j.Level.toLevel(logLevel)` also converts it to upper case internally.